app/models/post.rb

```diff
@@ -3,17 +3,26 @@ class Post < ApplicationRecord
   has_many :comments
   has_many :likes
 
-  def update_user_posts_counter
-    author.update(post_count: author.authored_posts.count)
-  end
+
 
   # Method to return the 5 most recent comments for a given post
   def recent_comments(limit = 5)
     comments.order(created_at: :desc).limit(limit)
   end
 
   # Method to update the likes counter for a post
+
+
+  after_save :update_user_posts_counter
+  after_save :update_likes_counter
+
+  private
+  def update_user_posts_counter
+    author.update(post_count: author.authored_posts.count)
+  end
+
   def update_likes_counter
     update(likes_counter: likes.count)
   end
```

Comment:
- [ ] You already have this method in the likes model, we don't want this function here as earlier reviewed.